samples/net/openthread/border_router/src/main.c

```diff
@@ -3,10 +3,141 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+#if (OTBR_ENABLE_WEB_SERVER == 1)
+    #include <zephyr/kernel.h>
+    #include <zephyr/net/net_if.h>
+    #include <zephyr/net/net_mgmt.h>
+    #include <zephyr/net/dhcpv4.h>
+    #include <zephyr/logging/log.h>
+    #include "web_server.h"
 
-int main(void)
-{
-	/* Nothing to do here. The Border Router is automatically started in the background. */
+    LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
 
-	return 0;
-}
+    static struct net_mgmt_event_callback net_cb;
+    static bool web_server_started = false;
+
+    static void net_event_handler(struct net_mgmt_event_callback *cb,
+                      uint64_t mgmt_event, struct net_if *iface)
+    {
+        LOG_INF("*** NETWORK EVENT: 0x%016llx ***", mgmt_event);
+
+        /* Print the event in a more readable format */
+        LOG_INF("Event details:");
+        LOG_INF("  - Event ID: 0x%016llx", mgmt_event);
+        LOG_INF("  - Interface: %p", iface);
+
+        /* Check for specific events we care about */
+        if (mgmt_event & NET_EVENT_IPV4_DHCP_BOUND) {
+            LOG_INF("  - DHCP BOUND detected!");
+            goto start_web_server;
+        }
+
+        if (mgmt_event & NET_EVENT_IPV4_ADDR_ADD) {
+            LOG_INF("  - IPv4 ADDR ADD detected!");
+            goto start_web_server;
+        }
+
+        if (mgmt_event & NET_EVENT_IPV6_ADDR_ADD) {
+            LOG_INF("  - IPv6 ADDR ADD detected!");
+            goto start_web_server;
+        }
+
+        /* Log all other events for debugging */
+        if (mgmt_event & NET_EVENT_IF_UP) {
+            LOG_INF("  - Interface UP");
+        }
+        if (mgmt_event & NET_EVENT_IF_DOWN) {
+            LOG_INF("  - Interface DOWN");
+        }
+        if (mgmt_event & NET_EVENT_IPV4_DHCP_START) {
+            LOG_INF("  - DHCP START");
+        }
+        if (mgmt_event & NET_EVENT_IPV4_DHCP_STOP) {
+            LOG_INF("  - DHCP STOP");
+        }
```

Comment:
I suggest you remove all these events if they are not used. And then this whole function can be refactored so that there is no need to have the goto either.